### ISSUE TITLE
How to clear DateTime field?

### ISSUE THREAD

**Issue by dmitriytretyakov** [laravel/nova-issues#2358]
I try to delete value from input and then click out of the field, system add some default value to field, how can i clear field?

**Comment by ziming** [comment]
It's a known bug currently.

**Comment by Equruler** [comment]
@davidhemphill 

Will you consider to make a "Clear" button to clear the field?

We sometimes define a field "valid_until" that allow user input empty value which is the meaning of "forever".

This bug is quite bothering.
Appreciate if you can handle this.

**Comment by zachgarcia1** [comment]
It seems if you set it to a date field you can clear it but date time won't let you.

**Comment by ziming** [comment]
Also is that it doesn't default to current date time value, only to current date value

**Comment by jbrooksuk** [comment]
I'm able to empty out a `DateTime` field, just by clearing its contents:

![Kapture 2020-03-03 at 12 49 17](https://user-images.githubusercontent.com/246103/75777218-7846eb00-5d4d-11ea-8bbd-c745e9dcce0b.gif)

**Comment by ziming** [comment]
@jbrooksuk 

After clearing the datetime field did you try to click on another field or elsewhere?

Or did you just go to click Update Post right away?

Because I just tested it and the value go back if i click on another field or the background. And I'm on the latest nova version.

**Comment by zachgarcia1** [comment]
yea it doesn't look like @jbrooksuk is on the latest version i'm currently using 2.12.0

**Comment by jbrooksuk** [comment]
> After clearing the datetime field did you try to click on another field or elsewhere in the background?
> 
> Or did you just go to click Update Post right away?
> 
> Because I just tested it again after your comment and the value go back if i click on another field or the background. And I'm on the latest nova version.

I shall look at this now.

> yea it doesn't look like @jbrooksuk is on the latest version i'm currently using 2.12.0

I only just released v2.12.0 😂

**Comment by zachgarcia1** [comment]
yep just sharing my version updated today @jbrooksuk

**Comment by zachgarcia1** [comment]
sorry for the slow erasing of fields just trying to show was i've been experiencing, and then like I said up top with the date field this will remove the value once its been set, it seems to only be an issue with the date time field

![datetime-field](https://user-images.githubusercontent.com/12938920/75792716-969bef00-5d2b-11ea-9162-5351d97c9b87.gif)

**Comment by dmitriytretyakov** [comment]
Yes, it's actual only for DateTime, my version is 2.12.0 and bug exists

**Comment by zachgarcia1** [comment]
@jbrooksuk where you able to reproduce this issue?

**Comment by michielkempen** [comment]
I'm currently on Laravel Nova v3.0.2 and I can confirm that this is a bug.

![Screen-Recording-2020-03-08-at-18 00 57](https://user-images.githubusercontent.com/14795113/76167452-0ca6b880-6167-11ea-86e5-610a0f695e29.gif)

**Comment by michielkempen** [comment]
Hey @jbrooksuk @davidhemphill, I spent some time debugging this issue and came up with a solution. 

There seem to be [a few open issues](https://github.com/flatpickr/flatpickr/issues?q=clear) on the flatpickr repository regarding this problem, but no clear solution. However, the problem can be easily overcome in Nova by applying the following changes to the `resources/js/components/DateTimePicker.vue` file.

```diff
<template>
  <input
    :disabled="disabled"
    :class="{ '!cursor-not-allowed': disabled }"
+   @keyup.delete="onClear"
    :value="value"
    ref="datePicker"
    type="text"
    :placeholder="placeholder"
  />
</template>

<script>
import flatpickr from 'flatpickr'
import 'flatpickr/dist/themes/airbnb.css'

export default {
  props: {
    ...
  },

  data: () => ({ flatpickr: null }),

  mounted() {
    this.$nextTick(() => {
      this.flatpickr = flatpickr(this.$refs.datePicker, {
        enableTime: this.enableTime,
        enableSeconds: this.enableSeconds,
        onClose: this.onChange,
        onChange: this.onChange,
        dateFormat: this.dateFormat,
        allowInput: true,
        // static: true,
        time_24hr: !this.twelveHourTime,
        locale: { firstDayOfWeek: this.firstDayOfWeek },
      })
    })
  },

  methods: {
    onChange(event) {
      this.$emit('change', this.$refs.datePicker.value)
    },
+   onClear(event) {
+     if(event.target.value === '') {
+       this.flatpickr.close();
+     }
+   }
  },

  beforeDestroy() {
    this.flatpickr.destroy()
  },
}
</script>

<style scoped>
.\!cursor-not-allowed {
  cursor: not-allowed !important;
}
</style>
```

The resulting workflow is as follows:

1. You can deselect/select a datetime as before.
2. Once you clear the input field using backspace or delete, the field is cleared and the datepicker is closed.

![test](https://user-images.githubusercontent.com/14795113/76361673-c5aff300-631f-11ea-864c-85b7d47c36ee.gif)

Any chance you could verify the solution and apply the fix?

Thanks in advance!

**Comment by jbrooksuk** [comment]
@michielkempen this didn't work for me using <kbd>Super+Del</kbd>.

**Comment by jbrooksuk** [comment]
However, it works if I select all and then delete, but that's only one way to delete it.

**Comment by dmitriytretyakov** [comment]
> However, it works if I select all and then delete, but that's only one way to delete it.

It's doesn't works for me

**Comment by Equruler** [comment]
How about make a "Clear" button, no matter on calendar or under the field?
Delete the text on a calendar field is not the obvious way, IMO.

And this approach should have no more arguing on deleting the text is work or not.

**Comment by dmitriytretyakov** [comment]
What about fix?

**Comment by zachgarcia1** [comment]
still not working on v3.1.0

**Comment by SuddenlyRust** [comment]
There is a workaround in the meantime https://github.com/laravel/nova-issues/issues/2344#issuecomment-602438340 @dmitriytretyakov @zachgarcia1 

It is working for me 🙏

**Comment by davidraijmakers** [comment]
This is still occuring

**Comment by dmitriytretyakov** [comment]
Please fix this

**Comment by zachgarcia1** [comment]
still not working on v3.2.1

**Comment by jbrooksuk** [comment]
Locking as we know this is a bug and is not yet fixed.

**Comment by davidhemphill** [comment]
Fixed via laravel/nova#907
